config/src/main/java/org/springframework/security/config/annotation/web/builders/HttpSecurity.java

```diff
@@ -1342,6 +1342,89 @@ public HttpSecurity authorizeHttpRequests(
 		return HttpSecurity.this;
 	}
 
+	/**
+	 * Allows restricting access based upon the {@link HttpServletRequest} using
+	 * {@link RequestMatcher} implementations (i.e. via URL patterns).
+	 *
+	 * <h2>Example Configurations</h2>
+	 *
+	 * The most basic example is to configure all URLs to require the role "ROLE_USER".
+	 * The configuration below requires authentication to every URL and will grant access
+	 * to both the user "admin" and "user".
+	 *
+	 * <pre>
+	 * &#064;Configuration
+	 * &#064;EnableWebSecurity
+	 * public class AuthorizeUrlsSecurityConfig extends WebSecurityConfigurerAdapter {
+	 *
+	 * 	&#064;Override
+	 * 	protected void configure(HttpSecurity http) throws Exception {
+	 * 		http
+	 * 			.authorizeHttpRequests((authorizeHttpRequests) ->
```

Comment:
Sorry, I should have been a bit clearer with my suggestion here. The code in the Javadoc should fit the method being described.

So, in this case, it would be:

```
http
    .authorizeHttpRequests()
        .antMatchers(&quot;/**&quot;).hasRoles(&quot;USER&quot;)
        .and()
    .formLogin();
```